benchmark/url/url-searchparams-iteration.js

```diff
@@ -0,0 +1,56 @@
+'use strict';
+const common = require('../common.js');
+const URLSearchParams = require('url').URLSearchParams;
+const v8 = require('v8');
+
+v8.setFlagsFromString('--allow_natives_syntax');
+
+const bench = common.createBenchmark(main, {
+  method: ['forEach', 'iterator'],
+  n: [1e6]
+});
+
+const str = 'one=single&two=first&three=first&two=2nd&three=2nd&three=3rd';
+
+function forEach(n) {
+  const params = new URLSearchParams(str);
+  const noop = () => {};
+  eval('%OptimizeFunctionOnNextCall(params.forEach)');
+  eval('%OptimizeFunctionOnNextCall(noop)');
+  params.forEach(noop);
+
+  bench.start();
+  for (var i = 0; i < n; i += 1)
+    params.forEach(noop);
```

Comment:
Added a `noDead` variable similar to your benchmark PR. Also note the increased performance of iterators as well, as iterators do not have a callback.